src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,142 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private static final int DIVIDE = 2;
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
     @Override
     public void add(T value) {
+        Node<T> lastNode = tail;
+        Node<T> newNode = new Node<>(lastNode, value, null);
+        tail = newNode;
+        if (lastNode == null) {
+            head = newNode;
+        } else {
+            lastNode.next = newNode;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index == size) {
+            add(value);
+        } else if (index == 0) {
+            head = new Node<>(null, value, head);
+            size++;
+        } else {
+            checkIndex(index);
+            Node<T> needNode = findingIndexNode(index);
+            Node<T> beforeNode = needNode.prev;
+            Node<T> newNode = new Node<>(beforeNode, value, needNode);
+            beforeNode.next = newNode;
+            needNode.prev = newNode;
+            size++;
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T element : list) {
+            add(element);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        return findingIndexNode(index).item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkIndex(index);
+        Node<T> oldNode = findingIndexNode(index);
+        T oldValue = oldNode.item;
+        oldNode.item = value;
+        return oldValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index);
+        return deleteNode(findingIndexNode(index));
     }
 
     @Override
     public boolean remove(T object) {
+        for (Node<T> needNode = head; needNode != null; needNode = needNode.next) {
+            if (needNode.item == object
+                    || needNode.item.equals(object)) {
+                deleteNode(needNode);
+                return true;
+            }
+        }
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private static class Node<T> {
+        private T item;
+        private Node<T> next;
+        private Node<T> prev;
+
+        public Node(Node<T> prev, T element, Node<T> next) {
+            this.prev = prev;
+            this.item = element;
+            this.next = next;
+        }
+    }
+
+    private Node<T> findingIndexNode(int index) {
+        Node<T> needNode;
```

Comment:
```suggestion
        Node<T> node;
```
When you have only one Node in the method you can call it a simple 'node'. When we have a few Node objects we should give them more specific names to avoid mess.